Readme update

amitschang opened the pull request:
This is an update to readme that comes with project goals, quickstart, link to documentation, a few notes and information for developers

fabbasinejad commented:
Would the goal of the beginign of the readme be such that someone can just copy past and try and run some example code?
if so: I wanted to suggest putting your code in one code block and adding your comments as to what is happening as actual comments. May be a preference thing. I like to be spoon fed :) 

Comment 2: should the setup section at the bottom come before the quick start?

fabbasinejad reviewed (changes requested):
Would the goal of the beginign of the readme be such that someone can just copy past and try and run some example code?
if so: I wanted to suggest putting your code in one code block and adding your comments as to what is happening as actual comments. May be a preference thing. I like to be spoon fed :)

Comment 2: should the setup section at the bottom come before the quick start?

amitschang commented:
@fabbasinejad, in a way, but also to walk someone through a most basic example. I could combine the top two blocks of python, but of course  I can't combine the pip command and the bottom block is sort of an alternative to the one above it (which is a bit why there are separated like that).

For the second comment, do you mean the developer docs? In general I think we should have user docs before developer docs in order. Or did you mean something else?

amitschang commented:
@fabbasinejad, thanks for the feedback. I added a TOC, combined two code blocks for a complete self-consistent example, and changed the developer section title (I think it was confusing, since pip install is all that is needed as a user and this is shown in the quick-start - the bottom section was just for development and from source-code). Thanks!

fabbasinejad reviewed (approved):
LGTM

amitschang commented:
Thanks!
